### PR TITLE
Port SRW App to WCOSS2

### PR DESCRIPTION
**Pull request by chan-hoo**
## DESCRIPTION OF CHANGES: 
- Port the UFS SRW App to WCOSS2 (Cactus / Dogwood).

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 
- WE2E tests on WCOSS2, Hera, and Orion:
MET_verification
community_ensemble_2mems_stoch
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_2017_gfdlmp_regional
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_HRRR
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15_thompson_mynn_lam3km
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_RRFS_v1beta
nco_grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
nco_grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_HRRR

- [x] hera.intel
- [x] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] gaea.intel
- [ ] jet.intel
- [x] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## ISSUE: 
Fixes issue mentioned in #365

## CHECKLIST
- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

## CONTRIBUTORS: 
@BenjaminBlake-NOAA

### CODE REVIEW

**Comment by chan-hoo** [comment]
@BenjaminBlake-NOAA @RatkoVasic-NOAA @MatthewPyle-NOAA @ShunLiu-NOAA @hu5970: Can you review and test this PR on Cactus?

**Comment by BenjaminBlake-NOAA** [comment]
@chan-hoo  Sure, I will test it on Cactus today.  Thanks for putting this together!

**Comment by MatthewPyle-NOAA** [comment]
Awesome on getting this together so quickly!  One quick question - what required the elimination of the module purge lines on Hera and Orion?

**Comment by chan-hoo** [comment]
@MatthewPyle-NOAA, as far as I know, 'module purge' does not work inside a module file.

File: parm/FV3LAM_wflow.xml
Comment (danielabdi-noaa) [inline parm/FV3LAM_wflow.xml]: Also there is a `SLURM_NATIVE_CMD` in machine file that you could set `-l place=excl` too. Name doesn't make sense here since wcoss uses pbspro, so maybe it can be changed to `SCHED_NATIVE_CMD`? I think the only place that is used currently is on Gaea.

File: ush/load_modules_run_task.sh
Comment (danielabdi-noaa) [inline ush/load_modules_run_task.sh:98]: Thanks for the fix!

File: parm/FV3LAM_wflow.xml
Comment (danielabdi-noaa) [inline parm/FV3LAM_wflow.xml]: I think this should be uppercase "WCOSS2"

**Review by danielabdi-noaa — COMMENTED** [review]
That looks good to me! I've left some minor suggestions

**Comment by ShunLiu-NOAA** [comment]
Thank you, Chan-Hoo.

Have you tested the workflow with CONUS-3km and the new North America 3km grid?

Shun

**Comment by chan-hoo** [comment]
@ShunLiu-NOAA, I tested it with CONUS 3km (grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15_thompson_mynn_lam3km), but I didn't test NA 3km. I'll run a WE2E test for NA 3km.

File: modulefiles/wflow_wcoss2
Comment (danielabdi-noaa) [inline modulefiles/wflow_wcoss2]: Do we need to put version numbers in `run.ver.wcoss2` and use them here? Maybe @MatthewPyle-NOAA knows.

**Comment by ShunLiu-NOAA** [comment]
Chan-Hoo,

If you can test C3359 and C3463 grid, it will be great. I assume conus-3km
is C3359 not C3357. They are the grid used for RRFS-A-CONUS and RRFS-A-NA.
Thank you.

Shun

**Comment by chan-hoo** [comment]
@ShunLiu-NOAA, The RRFS_CONUS_3km domain, used in WE2E, is C3359.

File: parm/FV3LAM_wflow.xml
Comment (chan-hoo) [reply parm/FV3LAM_wflow.xml]: You're correct. Updated!

File: parm/FV3LAM_wflow.xml
Comment (chan-hoo) [reply parm/FV3LAM_wflow.xml]: Renamed to SCHED_NATIVE_CMD!

File: modulefiles/wflow_wcoss2
Comment (MatthewPyle-NOAA) [reply modulefiles/wflow_wcoss2]: Good suggestion @danielabdi-noaa  - that is what the run.ver file is for (to hold version numbers in one centralized place).   Can you make that change, @chan-hoo ?

**Comment by MichaelLueken** [comment]
@chan-hoo Please update your `feature/port_wcoss2` branch to the latest in the official repository.  Several PRs were merged yesterday to add Jenkins testing functionality to the SRW.  Once this PR has received at least one approval from a code manager, the `run_we2e_fundamental_test` label will be added to kick off the Jenkins automated testing on all level 1 machines.  Once the Jenkins build and testing completes successfully, then this PR will be able to be merged.

**Comment by chan-hoo** [comment]
@MichaelLueken, Once a reviewer approves this PR, will the `run_we2e_fundamental_test` label be added automatically? or should one of the code reviewers add the label manually?

**Comment by MichaelLueken** [comment]
@chan-hoo Once a reviewer has approved this PR, then a code reviewer will need to add the label manually.

**Comment by chan-hoo** [comment]
I got it! I'll add the label unless the reviewers do.

**Review by BenjaminBlake-NOAA — APPROVED** [review]
I was able to build all components of the SRW app on Cactus and run the workflow successfully for the RRFS_CONUS_3km grid.  Since the WE2E tests passed for Chan-Hoo, I approve these changes.

**Comment by chan-hoo** [comment]
@MichaelLueken, the build failure on Gaea is not caused by this PR. I think @danielabdi-noaa opened a PR for this (#407).

**Comment by MichaelLueken** [comment]
@chan-hoo You are correct.  There was an update on Gaea, which is causing the failure on Jenkins for Gaea.  I will be moving forward with your changes once the rest of the tests pass.

**Review by MatthewPyle-NOAA — COMMENTED** [review]
Just realizing that my comment about inclusion of modules in run.ver was in limbo.  Submitting the comment now.

File: modulefiles/wflow_wcoss2
Comment (chan-hoo) [reply modulefiles/wflow_wcoss2]: Sure. Will do

**Comment by MichaelLueken** [comment]
@chan-hoo The Jenkins tests have completed.  Once you have addressed @MatthewPyle-NOAA and @danielabdi-noaa issue with using the versions in the run.ver file, I will be able to approve and commit this update.

**Review by danielabdi-noaa — APPROVED** [review]
Looks good to me!

**Pull request merged**
